torch/distributed/launch.py

```diff
@@ -0,0 +1,172 @@
+(4) In your training program, you are supposed to convert your model to
+DistributedDataParallel module using the following function. Please ensure
+that device_ids argument is set to be the only GPU device id that your code
+will be operating on. In other words, the device_ids needs to be [args.device]
+in order to use this utility.
+
+    model = torch.nn.parallel.DistributedDataParallel(model,
+                                                      device_ids=[arg.device])
+
+(5) For multi-node training, current we only support nodes with identical number
+of GPUs. In other words, the number of GPUs on each node needs to be the same.
+
+"""
+
+
+import subprocess
+import os
+import socket
+from argparse import ArgumentParser, REMAINDER
+
+import torch
+
+
+def parse_args():
+    """
+    Helper function parsing the command line options
+    @retval ArgumentParser
+    """
+    parser = ArgumentParser(description="PyTorch distributed training launch "
+                                        "helper utilty that will spawn up "
+                                        "multiple distributed processes")
+
+    # Optional arguments for the launch helper
+    parser.add_argument("--num_node", type=int, default=1,
+                        help="The number of nodes to use for distributed "
+                             "training")
+    parser.add_argument("--rank_node", type=int, default=0,
+                        help="The rank of the node for multi-node distributed "
+                             "training")
+    parser.add_argument("--nproc_per_node", type=int, default=-1,
+                        help="The number of processes to launch on each node, "
+                             "will default to the number of GPUs on your "
+                             "system if not specified")
+    parser.add_argument("--master_addr", default="127.0.0.1", type=str,
+                        help="Master node (rank 0)'s address, should be either "
+                             "the IP address or the hostname of node 0, for "
+                             "single node multi-proc training, the "
+                             "--master_addr can simply be 127.0.0.1")
+    parser.add_argument("--master_port", default=29500, type=int,
+                        help="Master node (rank 0)'s free port that needs to "
+                             "be used for communciation during distributed "
+                             "training")
+
+    # positional
+    parser.add_argument("training_script", type=str,
+                        help="The full path to the single GPU training "
+                             "program/script to be launched in parallel, "
+                             "followed by all the arguments for the "
+                             "training script")
+
+    # rest from the training program
+    parser.add_argument('training_script_args', nargs=REMAINDER)
+    return parser.parse_args()
+
+
+args = parse_args()
+num_gpus = torch.cuda.device_count()
```

Comment:
I'd really like this script to be independent of the GPUs in the system. You should be able to launch CPU-only jobs too (e.g. for RL)